File: lib_core/src/main/java/com/example/lib_core/http/observer/BaseObserver.java
package com.example.lib_core.http.observer;

import io.reactivex.observers.DisposableObserver;

//观察者
public abstract class BaseObserver<T> extends DisposableObserver<T> implements IObserver<T> {
    @Override
    public void onNext(T t) {
        success(t);
    }

    @Override
    public void onError(Throwable e) {
        error(e.getMessage());
    }

    @Override
    public void onComplete() {

    }
}

File: lib_core/src/main/java/com/example/lib_core/http/observer/BaseObserver.kt
package com.example.lib_core.http.observer

import io.reactivex.observers.DisposableObserver

//观察者
abstract class BaseObserver<T> : DisposableObserver<T>(), IObserver<T> {
    override fun onNext(t: T) {
        success(t)
    }

    override fun onError(e: Throwable) {
        error(e.message)
    }

    override fun onComplete() {}
}